client/report.go

```diff
@@ -0,0 +1,95 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+const reportEndpoint = "/transactions.xml"
+
+//ReportAppID - Report for the Application Id authentication pattern with serviceToken
+func (client *ThreeScaleClient) ReportAppID(appId string, token string, tokenType string, serviceId string, transactions ReportTransactions) (ApiResponse, error) {
```

Comment:
@PhilipGough I've introduced this concept of "tokenType" to switch between service_token and provider_key ... we will need to support both. ..

Reply:
right - how about wrapping that information in a struct containing both?

Reply:
That was my second thought about this, having an Auth struct containing the type and the token value.

```
 type TokenAuth struct {
 	Type   string
 	Value  string  
} 
```

Naming?

Reply:
`ProviderAuth`? `ServiceAuth`?

Reply:
What do you mean? two different structs? or one with two distinct strings? 

```
 type TokenAuth struct {
 	Type   string
 	ProviderAuth  string
       ServiceAuth string
} 
```

Reply:
I thought about one with the type and the value just like you proposed, provider auth and service auth would be two different names for the same struct (there is almost no difference for the 3scale backend client between both terms) that helps in identifying ourselves as having permission to call auth/rep on the service we care about.

Reply:
Having them as two separate types also gives us a way of "unwrapping" them in` url.Values` so +1 for this approach